iot-management-device/src/main/kotlin/com/example/iotmanagementdevice/kafka/consumer/DeviceUpdateProcessor.kt

```diff
@@ -25,23 +26,26 @@ class DeviceUpdateProcessor(
                 val updatedDevice = UpdateDeviceResponse.parser().parseFrom(record.value())
                 val notification: DeviceUpdateNotification =
                     deviceNotificationMapper.toDeviceUpdateNotification(updatedDevice)
-
                 sendNotification(notification)
                     .doFinally { record.receiverOffset().acknowledge() }
+            }.onErrorResume { error ->
+                log.error("Failed to process a device update message", error)
+                Unit.toMono()
```

Comment:
what will happen if parsing fails? Do you know if the next events will be consumed?